os/kernel/log_dump/log_dump_manager.c

```diff
@@ -0,0 +1,89 @@
+ *
+ * Copyright 2023 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/log_dump/log_dump_internal.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+/****************************************************************************
+ * Private Type Declarations
+ ****************************************************************************/
+/****************************************************************************
+ * Public Variables
+ ****************************************************************************/
+/****************************************************************************
+ * Private Variables
+ ****************************************************************************/
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+int log_dump_manager(int argc, char *argv[])
+{
+	if (log_dump_initialize() != LOG_DUMP_OK) {
+		lldbg("Fail to init log dump\n");
+		return 0;
+	}
+
+	struct log_dump_msg recv_msg;
+	int nbytes;
+	int compressed_size;
+	sem_t *prev_head_sem = NULL;
+
+	while (true) {
+		nbytes = mq_receive(mq_fd, (char *)&recv_msg, sizeof(struct log_dump_msg), NULL);
+		if (nbytes <= 0) {
+			continue;
+		}
+
+		int uncomp_size = UNCOMP_HEAD.curbytes;
+		
+		if (uncomp_size <= 0) {
+			lldbg("No log to compress");
+			UNCOMP_HEAD_NEXT();
+			sem_post(&UNCOMP_HEAD.buf_sem);
+			continue;
+		}
+
+		compressed_size = log_dump_compress(out_buf, UNCOMP_HEAD.arr, uncomp_size);
+		if (compressed_size == LOG_DUMP_COMPRESS_FAIL) {
+			lldbg("Fail to compress");
+		}
+		else {
+			log_dump_append((const char*)out_buf, compressed_size);
+		}
+
+		UNCOMP_HEAD.curbytes = 0;
+		
+		size_t free_size = log_dump_get_free_size();
```

Comment:
In this case, we don't need to declare local variable, because it is only used one-time.
If you use like `UNCOMP_HEAD.free_size = log_dump_get_free_size();` directly, you can save the stack.